### PR TITLE
Update haskell workflow

### PR DESCRIPTION
**Pull request by angerman**
Adjust to node 20 instead of 16.

### CODE REVIEW

File: .github/workflows/haskell.yml [hunk 1]
```diff
@@ -36,7 +37,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        ghc: ["8.10.7", "9.2.7", "9.6.4", "9.8.1"]
+        ghc: ["8.10.7", "9.6.4", "9.8.1"]
```
Comment (lehins) [inline .github/workflows/haskell.yml:40]: We still use it as our main development ghc version. I'd prefer not to remove it from CI for now. Also `9.8.2` is now available
```suggestion
        ghc: ["8.10.7", "9.2.8", "9.6.4", "9.8.2"]
```

**Review by lehins — COMMENTED** [review]
Other than list of GHC versions looks good

File: .github/workflows/haskell.yml (hunk 1 above)
Comment (angerman) [reply .github/workflows/haskell.yml:40]: We didn't move to 9.6? 👀

Comment (erikd) [reply .github/workflows/haskell.yml]: @lehins I think we did move to `9.6`.

Comment (lehins) [reply .github/workflows/haskell.yml]: Nope. Our main ghc for deployment of cardano-node is still 8.10.7

Comment (lehins) [reply .github/workflows/haskell.yml]: But for development, it depends on the repo. For ledger it is still 9.2.8

**Comment by angerman** [comment]
@lehins do you have admin to this repo? I don't seem to anymore.

**Review by lehins — APPROVED** [review]
LGTM. Thank you.

**Pull request merged**